kairon/shared/chat/agent/agent_flow.py

```
@@ -0,0 +1,287 @@
+                        'name': event.name,
+                        'type': event.type,
+                    } for event in rule.events]
+                return events, None
+            elif multiflow := MultiflowStories.objects(bot=self.bot, block_name=name).first():
+                events = multiflow.events
+                event_map, start = AgenticFlow.sanitize_multiflow_events(events)
+                return event_map, start
+            else:
+                raise DoesNotExist(f"[{name}] not found for this bot")
+        except DoesNotExist as e:
+            raise AppException(e)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)